src/AggregateFunctions/AggregateFunctionAnyBase.h

```diff
@@ -456,8 +331,8 @@ struct SingleValueDataString //-V730
 private:
     using Self = SingleValueDataString;
 
-    Int32 size = -1;    /// -1 indicates that there is no value.
-    Int32 capacity = 0;    /// power of two or zero
+    Int32 size = -1; /// -1 indicates that there is no value.
```

Comment:
Here `SingleValueDataString` is almost the same as in `AggregateFunctionMinMax`

Reply:
Yes, and breaking them apart to introduce improvements has been a huge pain. 

Improving the performance of ANY and ANYLAST is trivial (part of the work is done) but it's a huge pain to do everything at once . I would rather do this step by step: first optimize MIN/MAX, then ARGMIN/ARGMAX (I probably should have waited on this), then ANY/ANYLAST, and finally see what can be merged together. It's likely there could be a common base class, but any work I do today to keep MIN/MAX and ANY/ANYLAST will likely get in the way later.